tools/api-markdown-documenter/package.json

```diff
@@ -37,6 +37,7 @@
 		"test": "npm run test:mocha",
 		"test:coverage": "nyc npm test:mocha -- --reporter xunit --reporter-option output=nyc/junit-report.xml",
 		"test:mocha": "mocha dist/**/test/**/*.test.js --unhandled-rejections=strict --reporter-option maxDiffSize=100000",
+		"test:mocha:multireport": "npm run test:mocha -- --reporter mocha-multi-reporters --reporter-options configFile=testConfig.json",
```

Comment:
we should find a way to lean on existing extension of the common test config in `mocharc-common.js` rather than add another boilerplate `testConfig.json` file to each package.

Reply:
In the documentation for the mocha-multi-reporter package, it seems like it can only accept configurations in the form of a json file: https://www.npmjs.com/package/mocha-multi-reporters. I'm not sure if there is a way to work around this with the existing mocharc-common.js files.

Reply:
i wouldn't mind having a single json file in the `mocharc-common.js` package, which sounds like it should still be possible. Did you give modifying these lines in `packages/test/mocha-test-setup/mocharc-common.js` a try?

```js
			config["reporter-options"] = [
				`output=${packageDir}/nyc/junit-report.xml`,
				`suiteName=${packageJson.name}`,
			];
```

there's some examples in that file of injecting conditional config (see e.g. `FLUID_TEST_FORBID_ONLY`). You should be able to do something similar.

Reply:
This is a good callout here. Were you able to try this alternate approach here? If every single package needs testConfig.json, it may be hard to maintain it. Let's go down that route only if we do not have other choices.

Reply:
I've been testing this method out and it doesn't quite seem to work as expected. 
![image](https://user-images.githubusercontent.com/112977307/226470565-7e537d31-6735-4055-8b1c-20d958b3cbb6.png)
With this implementation, it seems that it'll look for a 'testConfig.json' within each mocha test directory instead of the mocharc-common.js directory. Creating a relative path wouldn't work either since there isn't a consistent folder structure across all mocha tests. I also tried absolute paths, but this method seems to fail when running on the pipeline since the root directory is different. Unless there's a way to point to a testConfig.json file relative to the mocharc-common.js directory, I'm not sure if this method will work well.

Reply:
JS generally has ways to resolve the location of the current running script: the `__dirname` global is probably the one which works here (which API you use is dependent on environment you're running in--`import.meta.url` is e.g. the api for js modules). So if `mocha-multi-reporters` demands a path to a file for its configuration, this should do about the right thing:

```
config["reporter-options"] = [`configFile=${path.join(__dirname, "test-config.json")}`]
```
you should check the escaping behavior of the above: I think it probably fails for people with spaces in their repository absolute path.